x-pack/test/rule_registry/spaces_only/tests/trial/lifecycle_executor.ts

```diff
@@ -0,0 +1,242 @@
+  createLifecycleExecutor,
+  WrappedLifecycleRuleState,
+} from '../../../../../plugins/rule_registry/server/utils/create_lifecycle_executor';
+import type { FtrProviderContext } from '../../../common/ftr_provider_context';
+import {
+  Dataset,
+  IRuleDataClient,
+  RuleDataService,
+} from '../../../../../plugins/rule_registry/server';
+import {
+  MockRuleParams,
+  MockRuleState,
+  MockAlertContext,
+  MockAlertState,
+  MockAllowedActionGroups,
+} from '../../../common/types';
+import { AlertExecutorOptions } from '../../../../../plugins/alerting/server';
+import { cleanupRegistryIndices } from '../../../common/lib/helpers/cleanup_registry_indices';
+
+// eslint-disable-next-line import/no-default-export
+export default function createLifecycleExecutorApiTest({ getService }: FtrProviderContext) {
+  // The getService('es') client returns the body of the transport requests.
+  // Where the client provided by Kibana returns the request response with headers, body, statusCode... etc.
+  // This cluster client will behave like the KibanaClient.
+  const es = getService('cluster_client');
+
+  const log = getService('log');
+
+  const fakeLogger = <Meta extends LogMeta = LogMeta>(msg: string, meta?: Meta) =>
+    meta ? log.debug(msg, meta) : log.debug(msg);
+
+  const logger = {
+    trace: fakeLogger,
+    debug: fakeLogger,
+    info: fakeLogger,
+    warn: fakeLogger,
+    error: fakeLogger,
+    fatal: fakeLogger,
+    log: sinon.stub(),
+    get: sinon.stub(),
+  } as Logger;
+
+  const getClusterClient = () => {
+    const client = es as ElasticsearchClient;
+    return Promise.resolve(client);
+  };
+
+  describe('createLifecycleExecutor', () => {
+    let ruleDataClient: IRuleDataClient;
+    before(async () => {
+      // First we need to setup the data service. This happens within the
+      // Rule Registry plugin as part of the server side setup phase.
+      const ruleDataService = new RuleDataService({
+        getClusterClient,
+        logger,
+        kibanaVersion: '8.0.0',
+        isWriteEnabled: true,
+        isWriterCacheEnabled: false,
+        disabledRegistrationContexts: [] as string[],
+      });
+
+      // This initializes the service. This happens immediately after the creation
+      // of the RuleDataService in the setup phase of the Rule Registry plugin
+      ruleDataService.initializeService();
+
+      // This initializes the index and templates and returns the data client.
+      // This happens in each solution plugin before they can register lifecycle
+      // executors.
+      ruleDataClient = ruleDataService.initializeIndex({
+        feature: AlertConsumers.OBSERVABILITY,
+        registrationContext: 'observability.test',
+        dataset: Dataset.alerts,
+        componentTemplateRefs: [],
+        componentTemplates: [
+          {
+            name: 'mappings',
+            mappings: mappingFromFieldMap(
+              {
+                testObject: {
+                  type: 'object',
```

Comment:
Awesome that you test here the `object` type. Once you PR is merged, I would be curious to test with the recently added `flattened` type as well.